src/owner.rs

```diff
@@ -0,0 +1,357 @@
+
+        #[derive(RustcDecodable)]
+        struct GithubTeam {
+            slug: String,
+            id: i32,
+        }
+
+        let json = try!(str::from_utf8(resp.get_body()).ok().chain_error(||{
+            internal("github didn't send a utf8-response")
+        }));
+        let teams: Vec<GithubTeam> = try!(json::decode(json).chain_error(|| {
+            internal("github didn't send a valid json response")
+        }));
+
+        let mut github_id = None;
+
+        for team in teams {
+            if team.slug == team_name {
+                github_id = Some(team.id);
+                break;
+            }
+        }
+
+        let github_id = try!(github_id.ok_or_else(|| {
+            human(format!("could not find the github team {}/{}", org_name, team_name))
+        }));
+
+        // mock Team (only need ID to check team status)
+        let team = Team { github_id: github_id, cargo_id: 0, name: String::new() };
+        if !try!(team.contains_user(req_user)) {
+            return Err(human("only members of a team can add it as an owner"));
+        }
+
+        // insert into DB for reals
+        try!(conn.execute("INSERT INTO teams
+                           (name, github_id)
+                           VALUES ($1, $2)",
+                          &[&name, &github_id]));
+
+        // read it right back out:
+        Team::find_by_name(conn, name)
+    }
+
+    /// Phones home to Github to ask if this User is a member of the given team.
+    /// Note that we're assuming that the given user is the one interested in
+    /// the answer. If this is not the case, then we could accidentally leak
+    /// private membership information here.
+    pub fn contains_user(&self, user: &User) -> CargoResult<bool> {
+        // GET teams/:team_id/memberships/:user_name
+        // check that "state": "active"
+
+        let resp = try!(http::handle()
+                         .get(format!("https://api.github.com/teams/{}/memberships/{}",
+                                   self.github_id, &user.gh_login))
+                         .header("Accept", "application/vnd.github.v3+json")
+                         .header("User-Agent", "hello!")
+                         .header("Authorization", &format!("token {}", &user.gh_access_token))
+                         .exec());
+
+        match resp.get_code() {
+            200 => {} // Ok!
+            404 => {
+                // Yes, this is actually how "no membership" is signaled
+                return Ok(false);
+            }
+            403 => {
+                return Err(human("It looks like you don't have permission \
+                                  to query an organization that owns this \
+                                  crate. You may need to re-authenticate on \
+                                  crates.io to grant permission to read \
+                                  github org memberships. Just go to \
+                                  https://crates.io/login"));
```

Comment:
I think it's kinda important from a UX perspective -- people are going to try this feature and get walled on out-of-date authentication. There's two places where this can happen, and they're sufficiently different that they warrant different messages IMO.